contracts/strategies/morpho/MetaMorphoStrategy.sol

```diff
@@ -149,11 +187,104 @@ contract MetaMorphoStrategy is ERC4626Strategy {
 
   /// @dev return always a value which is multiplied by 1e18
   /// eg for 2% apr -> 2*1e18
-  /// @notice this lending market is returning the apr already compounded (apy)
   function getApr() external view override returns (uint256 apr) {
+    apr = getAprWithLiquidityChange(0, 0);
+  }
+
+  /// @dev return always a value which is multiplied by 1e18
+  /// eg for 2% apr -> 2*1e18. Add and sub params are used off-chain
+  /// to calculate the variation of the apr based on the liquidity added/removed
+  function getAprWithLiquidityChange(uint256 add, uint256 sub) public view returns (uint256 apr) {
+    // TODO update mm snippet to include add and sub
+    // uint256 ratePerSecond = IMetamorphoSnippets(mmSnippets).supplyAPYVault(strategyToken, add, sub);
+
     uint256 ratePerSecond = IMetamorphoSnippets(mmSnippets).supplyAPYVault(strategyToken);
     // ratePerSecond is the rate per second scaled by 18 decimals
-    // (eg 32943060 -> 32943060 * 24 * 3600 * 365 * 100 / 1e18 = 0.103% apy)
-    apr = ratePerSecond * 365 days * 100;
+    // (eg 32943060 -> 32943060 * 24 * 3600 * 365 * 100 / 1e18 = 0.103% apr)
+    apr = ratePerSecond * 365 days * 100 + getRewardsApr(add, sub);
+  }
+
+  /// @notice return the additional rewards apr
+  /// @param add liquidity to add
+  /// @param sub liquidity to remove
+  /// @dev return a value multiplied by 1e18 eg for 2% apr -> 2*1e18. Add and sub params
+  /// are used off-chain to calculate the variation of the apr based on the liquidity added/removed
+  function getRewardsApr(uint256 add, uint256 sub) public view returns (uint256 apr) {
+    IMMVault _mmVault = IMMVault(address(strategyToken));
+    AprData memory _aprData = AprData(add, sub, _mmVault.withdrawQueueLength(), _mmVault.totalAssets());
+    address[] memory _rewardTokens = rewardTokens;
+    RewardData memory _rewardData;
+
+    for (uint256 i = 0; i < _rewardTokens.length; i++) {
+      _rewardData = rewardsData[_rewardTokens[i]];
+      if (_rewardData.sender == address(0) || (_rewardData.rewardToken == MORPHO && !morphoTransferable)) {
```

Comment:
why is `_rewardData.sender` checked not be `address(0)` but not the other params?

Reply:
clear thx